_search-plugins/sql/ppl/functions.md

```diff
@@ -771,3 +771,198 @@ sepal_length_in_cm | sepal_width_in_cm | petal_length_in_cm | petal_width_in_cm
+`input-field` | A field in `lookup-index` where matched values are applied to result output. You can specify multiple `input-field` values with commas. If you don't specify any `input-field`, all fields except `lookup-mapping-field` from `lookup-index` where matched values are applied to result output. | No |
+`output-field` | A field of output. You can specify zero or multiple `output-field` values. If you specify `output-field` with an existing field name in source query, its values will be replaced or appended by matched values from `input-field`. If the field specified in `output-field` is a new field, an extended new field will be applied to the results. | No | `input-field`
+
+The following examples use the `workers` and `work_information` indexes:
+
+`workers`:
+
+| id | name | occupation | country | salary
+:--- | :--- | :--- | :--- | :---
+| 1000 | Jake | Engineer | England | 100000
+| 1001 | Hello | Artist | USA | 70000
+| 1002 | John | Doctor | Canada | 120000
+| 1003 | David | Doctor | - | 120000
+| 1004 | David | - | Canada | 0
+| 1005 | Jane | Scientist | Canada | 90000
+
+`work_information`:
+
+| uid | name  | department | occupation
+:--- | :--- | :--- | :---
+| 1000 | Jake  | IT | Engineer |
+| 1002 | John  | DATA | Scientist |
+| 1003 | David | HR | Doctor |
+| 1005 | Jane  | DATA | Engineer |
+| 1006 | Tom   | SALES | Artist |
+
+**Example 1: Lookup workers and return the corresponding department**
+
+The following example looks up workers and returns the corresponding department:
+
+```sql
+source = workers | lookup work_information uid as id append department
+```
+
+| id | name | occupation | country | salary | department
+:--- | :--- | :--- | :--- | :--- | :---
+1000 | Jake | Engineer | England | 100000 | IT
+1001 | Hello | Artist | USA | 70000 | null
+1002 | John | Doctor | Canada | 120000 | DATA
+1003 | David | Doctor | null | 120000 | HR
+1004 | David | null | Canada | 0 | null
+1005 | Jane | Scientist | Canada | 90000 | DATA
+
+
+**Example 2: Lookup workers and replace occupation and department**
+
+The following example looks up workers and replaces occupation and department from work_information:
+
+```sql
+source = workers | lookup work_information uid as id, name
+```
+
+| id | name | occupation | country | salary | department
+:--- | :--- |:-----------| :--- | :--- | :---
+1000 | Jake | Engineer   | England | 100000 | IT
+1001 | Hello | null       | USA | 70000 | null
+1002 | John | Scientist  | Canada | 120000 | DATA
+1003 | David | Doctor     | null | 120000 | HR
+1004 | David | null       | Canada | 0 | null
+1005 | Jane | Engineer  | Canada | 90000 | DATA
+
+**Example 3: Lookup workers and create a new occupation field**
+
+The following example looks up workers and appends occupation from work_information as a new field:
+
+```sql
+source = workers | lookup work_information name replace occupation as new_occupation
+```
+
+| id | name | occupation | country | salary | new_occupation
```

Comment:
Please correct the capitalization and capitalize "Null" in the table.

Reply:
These are meant to be case accurate to the code, because identifiers in PPL are case-sensitive.